pkg/kubelet/winstats/winstats.go

```diff
@@ -103,9 +105,24 @@ func (c *StatsClient) WinContainerInfos() (map[string]cadvisorapiv2.ContainerInf
 }
 
 // WinMachineInfo returns a cadvisorapi.MachineInfo with details about the
-// node machine. Analogous to cadvisor MachineInfo method.
+// node machine. Run the powershell command to get the SystemUUID for Windows node
+// in here if it isn't provided by cadvisor.
 func (c *StatsClient) WinMachineInfo() (*cadvisorapi.MachineInfo, error) {
-	return c.client.getMachineInfo()
+	infos, err := c.client.getMachineInfo()
+	if err != nil {
+		return nil, err
+	}
+
+	if infos.SystemUUID == "" {
```

Comment:
System UUID shoudln't change throughout the the lifetime of kubelet. Running in once and just return the in-memory copy should be sufficient.

In fact, I think this (fields not changing) applies to most of fields in MachineInfo too.  + @dashpole to confirm the whether that's the case :-)

Reply:
I believe the kubelet actually does caching _on top of_ what is returned by the stats library.  https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/kubelet_getters.go#L357.  I think the library should fetch the whole thing on each call.

Reply:
You're right. I forgot that it's kubelet that cached this. Thanks!